nsv13/game/lazylift.dm

```diff
@@ -0,0 +1,394 @@
+
+//Lifts speak a lotta languages, alien isn't one of 'em.
+/datum/language_holder/synthetic/turbolift
+	languages = list(/datum/language/common)
+	shadow_languages = list(/datum/language/common, /datum/language/machine, /datum/language/draconic, /datum/language/drone)
+
+/obj/machinery/lazylift
+	name = "Turbolift interface panel"
+	desc = "A control panel for elevators, simply talk into its microphone to tell it where you want to go. Brought to you by the Sirius Cybernetics Corporation."
+	icon = 'nsv13/icons/obj/computers.dmi'
+	icon_state = "turbolift"
+	pixel_y = 26 //E.
+	anchored = TRUE
+	can_be_unanchored = FALSE
+	movement_type = FLYING //Turbolifts can't fall for anyone - won't get fooled again like I did.
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF //Go away greytiders
+	var/turf_type = null
+	var/deck = 1 //For display and record-keeping purposes.
+	var/obj/machinery/lazylift/master/master = null
+	var/list/platform = list() //The """platform""" of the lift that's going to move up and down. This is just a list of turfs that we own.
+	var/list/doors = list()
+	var/turbolift_delay = 4 SECONDS //How long should the turbolift stay on each deck? Don't make this delay higher than a few seconds, or BYOND will start to complain.
+	var/wait_time = 5 SECONDS //Brief cooldown after the lift reaches its destination, to allow people from that floor to board it.
+	var/list/moving_blacklist = list(/obj/machinery/lazylift, /obj/machinery/lazylift/master, /obj/machinery/light, /obj/structure/cable, /obj/machinery/power/apc, /obj/machinery/door, /obj/machinery/airalarm, /obj/machinery/firealarm)
+
+	//Voice activation.
+	flags_1 = HEAR_1
+	initial_language_holder = /datum/language_holder/synthetic/turbolift
+	var/list/addresses = list() //Voice activation! Lets you speak into the elevator to tell it where you wanna go. This stores all the departments on this floor. If youre lazy and re-use floors then uh...sucks to be you I guess!
+	var/list/languages = list(/datum/language/common, /datum/language/drone, /datum/language/machine, /datum/language/draconic)
+	var/list/area_blacklist = list(/area/space, /area/shuttle/turbolift, /area/shuttle, /area/maintenance/ship_exterior) //Areas that do not show up on the address book.
+	var/next_voice_activation = 0
+
+/obj/machinery/lazylift/attack_robot(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/lazylift/attack_ai(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/lazylift/attack_hand(mob/user)
+	var/list/deckNums = list()
+	for(var/obj/machinery/lazylift/LL in master.decks)
+		deckNums += LL.deck
+	var/theDeck = input(user, "Which deck would you like to go to?", name, null) as null|anything in deckNums
+	if(!theDeck || !isnum(theDeck))
+		return FALSE
+	master.path_to(theDeck)
+
+/obj/machinery/lazylift/Hear(message, atom/movable/speaker, message_language, raw_message, radio_freq, list/spans, message_mode)
```

Comment:
Have you thought about using signals and the voice analyser object for this to a degree?

Reply:
So this is actually from voice analyser code, using signals would make it more complicated as I'd have to register a speech signal on any mob that walks in to the elevator

Reply:
That makes sense. We might want to make a Hear component then.